front/src/modules/rollingStock/components/RollingStockCard/RollingStockCardDetail.tsx

```diff
@@ -18,15 +22,18 @@ type RollingStockCardDetailProps = {
   setCurvesComfortList: (curvesComfortList: RollingStockComfortType[]) => void;
 };
 
-export const listCurvesComfort = (curvesData: RollingStock['effort_curves']) => {
-  const comfortList: RollingStockComfortType[] = ['STANDARD'];
-  Object.keys(curvesData.modes).forEach((mode) => {
-    curvesData.modes[mode].curves.forEach((curve) => {
-      if (curve.cond?.comfort) {
-        if (!comfortList.includes(curve.cond.comfort)) comfortList.push(curve.cond.comfort);
+export const getCurvesComforts = (curvesData: EffortCurveForms) => {
+  const comfortList: RollingStockComfortType[] = [STANDARD_COMFORT_LEVEL];
+
+  Object.keys(curvesData).forEach((mode) => {
+    curvesData[mode].curves.forEach((curve) => {
+      const { comfort } = curve.cond;
+      if (comfort) {
+        if (!comfortList.includes(comfort)) comfortList.push(comfort);
```

Comment:
Could not use a map here as a map will return the same number of elements, but this would mean we got duplicated elements. But here we only add in the array the elements which are not already in it (so at the end, no duplicates).
We can use a reduce if you prefer (see code change)